api/src/booking/booking.service.ts

```diff
@@ -41,7 +41,7 @@ export class BookingService {
     let location: LocationEntity;
     if (createDto.locationName) {
       location = await this.locationRepository.findOne({
```

Comment:
maybe use `findOneOrFail` to avoid the null location.

Reply:
I keep location as optional, so, If fail, it will not block booking creation as current solution, but we can get confirmation from Graeme

Reply:
sounds good!